core/src/Cuda/Kokkos_Cuda_BlockSize_Deduction.hpp

```diff
@@ -198,6 +198,31 @@ int cuda_get_opt_block_size(const CudaInternal* cuda_instance,
                                 LaunchBounds{});
 }
 
+// Assuming cudaFuncSetCacheConfig(MyKernel, cudaFuncCachePreferL1)
+inline size_t get_shmem_per_sm_prefer_l1(cudaDeviceProp const& properties) {
+  int const compute_capability = properties.major * 10 + properties.minor;
+  return [compute_capability]() {
+    switch (compute_capability) {
+      case 30:
+      case 32:
+      case 35: return 16;
+      case 37: return 80;
+      case 50:
+      case 53:
+      case 60:
+      case 62: return 64;
+      case 52:
+      case 61: return 96;
+      case 70:
+      case 80: return 8;
+      case 75: return 32;
+      default:
+        Kokkos::Impl::throw_runtime_exception(
+            "Unknown device in cuda block size deduction");
+    }
```

Comment:
Two things: 

- `[[fallthrough]]` (macro-protected as something like `KOKKOS_IMPL_ATTRIBUTE_FALLTHROUGH`
- cite where these numbers come from in a (set of) comments.

Reply:
> `[[fallthrough]]` (macro-protected as something like KOKKOS_IMPL_ATTRIBUTE_FALLTHROUGH

The `[[fallthrough]]` attribute is a C++17 facility.  Are you sure compilers will issue a warning with empty case labels?  In any case I'd rather not solve this problem (if any) in this PR.

> cite where these numbers come from in a (set of) comments.

Will do.